src/aggregate/aggregate_request.c

```diff
@@ -273,13 +273,11 @@ static int handleCommonArgs(ParseAggPlanContext *papCtx, ArgsCursor *ac, QueryEr
       REQFLAGS_AddFlags(papCtx->reqflags, QEXEC_F_SEND_NOFIELDS);
       // TODO: unify if when req holds only maxResults according to the query type.
       //(SEARCH / AGGREGATE)
-    } else if ((arng->limit > *papCtx->maxSearchResults) &&
-               (*papCtx->reqflags & QEXEC_F_IS_SEARCH)) {
+    } else if ((arng->limit > *papCtx->maxSearchResults) && (*papCtx->reqflags & (QEXEC_F_IS_SEARCH | QEXEC_F_IS_HYBRID_SEARCH_SUBQUERY))) {
```

Comment:
Can't we use `IsSearch()` and `sHybridSearchSubquery()`?

```suggestion
    } else if ((arng->limit > *papCtx->maxSearchResults) && (IsSearch(papCtx) || IsHybridSearchSubquery(papCtx))) {
```